File: customer-module/frontend/src/app/order.service.ts
import { HttpClient } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root',
})
export class OrderService {
  private baseUrl = 'https://pos-customer.herokuapp.com/customer/';
  constructor(private httpClient: HttpClient) {}

  public getOrders(customerId): Observable<any> {
    return this.httpClient.get(this.baseUrl + 'orders/' + customerId);
  }

  public getOrder(orderId): Observable<any> {
    return this.httpClient.get(this.baseUrl + 'order/' + orderId);
  }
}

File: customer-module/backend/src/main/java/com/pos/customer/service/OrderService.java
package com.pos.customer.service;

import org.springframework.http.ResponseEntity;

import com.pos.customer.dto.OrderListDto;
import com.pos.customer.entity.Order;

public interface OrderService {
	String baseUrl="https://admin-pos.herokuapp.com/api/";
	ResponseEntity<OrderListDto> getOrders(Long customerId) throws Exception;
	ResponseEntity<Order> getOrder(Long orderId) throws Exception ;
}
